tests/core/test_variable.py

```diff
@@ -0,0 +1,548 @@
+    column = IntCol
+    entity = Person
+    definition_period = MONTH
+    label = u"Variable with stop attribute only coming before formula start."
+    end = '1990-01-01'
+
+    def formula_2000_01_01(self, individu, period):
+        return vectorize(self)
+
+
+def test_add__stop_attribute_before__one_formula__start():
+    check_error_at_add_variable(tax_benefit_system, stop_attribute_before__one_formula__start, 'You declared that "stop_attribute_before__one_formula__start" ends on "1990-01-01", but you wrote a formula to calculate it from "2000-01-01"')
+
+
+# end, formula with dates intervals overlap.
+
+class end_attribute_restrictive__one_formula(Variable):
+    column = IntCol
+    entity = Person
+    definition_period = MONTH
+    label = u"Variable with end attribute, one dated formula and dates intervals overlap."
+    end = '2001-01-01'
+
+    def formula_2001_01_01(self, individu, period):
+        return vectorize(self, 100)
+
+
+tax_benefit_system.add_variable(end_attribute_restrictive__one_formula)
+
+
+def test_call__end_attribute_restrictive__one_formula():
+    month = '2000-12'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute_restrictive__one_formula', month) == IntCol.default
+
+    month = '2001-01'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute_restrictive__one_formula', month) == 100
+
+    month = '2000-05'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute_restrictive__one_formula', month) == IntCol.default
+
+
+# formulas of different names (without dates overlap on formulas)
+
+class end_attribute__formulas__different_names(Variable):
+    column = IntCol
+    entity = Person
+    definition_period = MONTH
+    label = u"Variable with end attribute, multiple dated formulas with different names."
+    end = '2010-12-31'
+
+    def formula_2000_01_01(self, individu, period):
+        return vectorize(self, 100)
+
+    def formula_2005_01_01(self, individu, period):
+        return vectorize(self, 200)
+
+    def formula_2010_01_01(self, individu, period):
+        return vectorize(self, 300)
+
+
+tax_benefit_system.add_variable(end_attribute__formulas__different_names)
+
+
+def test_call__end_attribute__formulas__different_names():
+    month = '2000-01'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute__formulas__different_names', month) == 100
+
+    month = '2005-01'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute__formulas__different_names', month) == 200
+
+    month = '2010-12'
+    simulation = new_simulation(tax_benefit_system, month)
+    assert simulation.calculate('end_attribute__formulas__different_names', month) == 300
+
+
+def test_clone__end_attribute__formulas__different_names():
```

Comment:
>In the future, please try to make your tests more focused.
>
>It's ok do add all sorts of assertions in the code when developing, if it can help you understand the complicated and lopsided structure of OpenFisca.

>But keep in mind that _every line of code_ added to `master`, including tests, will have to be maintained and understood by OpenFisca developers.

> Also, we want to change deeply the architecture of `core`. Over-testing it will have a cost : if our tests tend to check more than they should do, _every time_ we will change something (for instance renaming`IntCol`), dozens of tests that have _nothing to do_ with what we are doing will break.